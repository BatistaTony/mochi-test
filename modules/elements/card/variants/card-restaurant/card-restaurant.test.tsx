import { act, fireEvent, render, screen, waitFor } from '@test-utils';

import CardRestaurant from './index';

describe('Restaurant card testing', () => {
  it('properly renders the mini restaurant card', async () => {
    const storeName = 'store name';
    const { container } = render(
      <CardRestaurant
        compressed
        image={{ small: 'img-300w.png', medium: 'img-600w.png' }}
        onClick={jest.fn()}
        logo="/img-300w.png"
        storeName={storeName}
        isOpen
        limitHour={10}
        distanceMin={10}
        likes={10}
      />,
      { newTheme: true }
    );

    await waitFor(() =>
      expect(screen.getByRole('article', { name: /restaurant/i })).toBeInTheDocument()
    );
    await waitFor(() =>
      expect(screen.getByRole('figure', { name: storeName })).toBeInTheDocument()
    );
    await waitFor(() => expect(screen.getByAltText(storeName)).toBeInTheDocument());
    await waitFor(() => expect(screen.getByAltText(`${storeName} logo`)).toBeInTheDocument());
    await waitFor(() => expect(screen.getByText(storeName)).toBeInTheDocument());

    expect(container).toMatchSnapshot();
  });
  it('properly renders the restaurant card', async () => {
    const storeName = 'store name';
    const { container } = render(
      <CardRestaurant
        onClick={jest.fn()}
        logo="/img-300w.png"
        storeName="store name"
        isOpen
        limitHour={10}
        distanceMin={10}
        likes={10}
        image={{ small: 'img-300w.png', medium: 'img-600w.png' }}
      />,
      { newTheme: true }
    );

    await waitFor(() =>
      expect(screen.getByRole('article', { name: /restaurant/i })).toBeInTheDocument()
    );
    await waitFor(() =>
      expect(screen.getByRole('figure', { name: storeName })).toBeInTheDocument()
    );
    await waitFor(() => expect(screen.getByAltText(storeName)).toBeInTheDocument());
    await waitFor(() => expect(screen.getByAltText(`${storeName} logo`)).toBeInTheDocument());
    await waitFor(() => expect(screen.getByText(storeName)).toBeInTheDocument());
    expect(container).toMatchSnapshot();
  });
  it('allows restaurant card click', async () => {
    const onClick = jest.fn();
    render(
      <CardRestaurant
        isOpen
        onClick={onClick}
        logo="/img-300w.png"
        storeName="store name"
        limitHour={10}
        distanceMin={10}
        likes={10}
        image={{ small: 'img-300w.png', medium: 'img-600w.png' }}
      />,
      { newTheme: true }
    );

    await act(async () => {});

    const card = screen.getByRole('article', { name: /restaurant/i });
    expect(card).toBeInTheDocument();

    fireEvent.click(card);
    await waitFor(() => expect(onClick).toHaveBeenCalledTimes(1));
  });
});
