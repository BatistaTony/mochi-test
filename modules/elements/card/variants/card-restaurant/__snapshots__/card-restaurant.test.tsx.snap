// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Restaurant card testing properly renders the mini restaurant card 1`] = `
.c2 {
  width: 100%;
  height: 100%;
  position: relative;
}

.c2 .lazyload,
.c2 .lazyloading {
  opacity: 0;
}

.c2 .lazyload,
.c2 .lazyloaded,
.c2 {
  opacity: 1;
  -webkit-transition: 2s cubic-bezier(0.215,0.61,0.355,1);
  transition: 2s cubic-bezier(0.215,0.61,0.355,1);
}

.c2:before {
  content: '';
  position: absolute;
  z-index: -1;
  top: 50%;
  left: 50%;
  margin-top: -16px;
  margin-left: -16px;
  width: 32px;
  height: 32px;
  border-radius: 50%;
  -webkit-animation: fTPTbj 1.2s infinite ease-in-out;
  animation: fTPTbj 1.2s infinite ease-in-out;
  background-color: #745CA0;
}

.c2 img {
  position: relative;
  width: 100%;
  max-height: 100%;
  z-index: 0;
  object-fit: contain;
}

.c3 {
  overflow: hidden;
  z-index: 1;
  background-color: #FAFCFE;
  margin-top: -9rem;
  border-radius: 50%;
  height: 15rem;
  width: 15rem;
  box-shadow: 0 0 0.5rem rgba(0,0,0,0.3);
}

.c0 {
  width: 100%;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  height: 10rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.5rem;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  margin-top: 0.5rem;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c10 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c11 {
  margin: 0.5rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #19795B;
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
  padding-left: 1rem;
  padding-right: 1rem;
  border-radius: 0.937rem;
}

.c13 {
  font-size: 1.6rem;
  margin: 0.5rem;
  color: #D7263D;
}

.c8 {
  font-size: 1.6rem;
  margin: 0.5rem;
  color: rgba(0,0,0,0);
  font-weight: bold;
}

.c8 path {
  stroke: #33393F;
  stroke-width: 5rem;
  stroke-linejoin: round !important;
  stroke-linecap: round !important;
}

.c5 {
  line-height: 120%;
  -webkit-letter-spacing: calc(2.8rem * -0.02);
  -moz-letter-spacing: calc(2.8rem * -0.02);
  -ms-letter-spacing: calc(2.8rem * -0.02);
  letter-spacing: calc(2.8rem * -0.02);
  font-size: 2.8rem;
}

.c9 {
  line-height: 150%;
  font-size: 1.4rem;
  color: #475058;
}

.c12 {
  line-height: 150%;
  font-size: 1.4rem;
  text-transform: capitalize;
  color: #FFFFFF;
}

<div>
  <div
    id="__next"
  >
    <div
      aria-label="restaurant"
      class="c0"
      display="flex"
      role="article"
      width="100%"
    >
      <div
        aria-label="store name"
        class="c1"
        display="flex"
        height="10rem"
        role="figure"
      >
        <div
          class="c2"
        >
          <img
            alt="store name"
            class="lazyload"
            data-sizes="auto"
            data-srcset="S3_BUCKET/img-300w.png 300w, S3_BUCKET/img-600w.png 600w"
            src="S3_BUCKET/img-600w.png"
            srcset="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
          />
        </div>
      </div>
      <div
        aria-label="store name logo"
        class="c3"
        height="15rem"
        role="figure"
        width="15rem"
      >
        <div
          style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
        >
          <div
            style="box-sizing: border-box; display: block; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTUwIiBoZWlnaHQ9IjE1MCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
              style="max-width: 100%; display: block;"
            />
          </div>
          <img
            alt="store name logo"
            decoding="async"
            src="/_next/image?url=%2Fimg-300w.png&w=640&q=75"
            srcset="/_next/image?url=%2Fimg-300w.png&w=256&q=75 1x, /_next/image?url=%2Fimg-300w.png&w=384&q=75 2x, /_next/image?url=%2Fimg-300w.png&w=640&q=75 3x"
            style="visibility: visible; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </div>
      </div>
      <div
        class="c4"
        display="flex"
        width="100%"
      >
        <p
          class="c5"
        >
          store name
        </p>
        <div
          class="c6"
          display="flex"
          width="100%"
        >
          <div
            aria-label="distance in minutes"
            class="c7"
            display="flex"
          >
            <svg
              aria-label="location"
              class="c8"
              fill="currentColor"
              height="1em"
              role="img"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M444.52 3.52L28.74 195.42c-47.97 22.39-31.98 92.75 19.19 92.75h175.91v175.91c0 51.17 70.36 67.17 92.75 19.19l191.9-415.78c15.99-38.39-25.59-79.97-63.97-63.97z"
              />
            </svg>
            <p
              aria-label="time"
              class="c9"
              color="grayQuaternary"
            >
              10
              min
            </p>
          </div>
          <div
            class="c10"
            display="inline-flex"
          >
            <div
              class="c11"
              display="inline-flex"
            >
              <p
                class="c12"
                color="textInverted"
              >
                open
              </p>
            </div>
            <p
              class="c9"
              color="grayQuaternary"
            >
              until 10h
            </p>
          </div>
          <div
            aria-label="like counter"
            class="c7"
            display="flex"
          >
            <svg
              aria-label="likes"
              class="c13"
              fill="currentColor"
              height="1em"
              role="img"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
              />
            </svg>
            <p
              class="c9"
              color="grayQuaternary"
            >
              10
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Restaurant card testing properly renders the restaurant card 1`] = `
.c2 {
  width: 100%;
  height: 100%;
  position: relative;
}

.c2 .lazyload,
.c2 .lazyloading {
  opacity: 0;
}

.c2 .lazyload,
.c2 .lazyloaded,
.c2 {
  opacity: 1;
  -webkit-transition: 2s cubic-bezier(0.215,0.61,0.355,1);
  transition: 2s cubic-bezier(0.215,0.61,0.355,1);
}

.c2:before {
  content: '';
  position: absolute;
  z-index: -1;
  top: 50%;
  left: 50%;
  margin-top: -16px;
  margin-left: -16px;
  width: 32px;
  height: 32px;
  border-radius: 50%;
  -webkit-animation: fTPTbj 1.2s infinite ease-in-out;
  animation: fTPTbj 1.2s infinite ease-in-out;
  background-color: #745CA0;
}

.c2 img {
  position: relative;
  width: 100%;
  max-height: 100%;
  z-index: 0;
  object-fit: contain;
}

.c3 {
  overflow: hidden;
  z-index: 1;
  background-color: #FAFCFE;
  margin-top: -7.5rem;
  border-radius: 50%;
  height: 15rem;
  width: 15rem;
  box-shadow: 0 0 0.5rem rgba(0,0,0,0.3);
}

.c0 {
  width: 100%;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  height: 15rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 2rem;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  margin-top: 2rem;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c10 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c11 {
  margin: 0.5rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #19795B;
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
  padding-left: 1rem;
  padding-right: 1rem;
  border-radius: 0.937rem;
}

.c13 {
  font-size: 1.6rem;
  margin: 0.5rem;
  color: #D7263D;
}

.c8 {
  font-size: 1.6rem;
  margin: 0.5rem;
  color: rgba(0,0,0,0);
  font-weight: bold;
}

.c8 path {
  stroke: #33393F;
  stroke-width: 5rem;
  stroke-linejoin: round !important;
  stroke-linecap: round !important;
}

.c5 {
  line-height: 120%;
  -webkit-letter-spacing: calc(2.8rem * -0.02);
  -moz-letter-spacing: calc(2.8rem * -0.02);
  -ms-letter-spacing: calc(2.8rem * -0.02);
  letter-spacing: calc(2.8rem * -0.02);
  font-size: 2.8rem;
}

.c9 {
  line-height: 150%;
  font-size: 1.4rem;
  color: #475058;
}

.c12 {
  line-height: 150%;
  font-size: 1.4rem;
  text-transform: capitalize;
  color: #FFFFFF;
}

<div>
  <div
    id="__next"
  >
    <div
      aria-label="restaurant"
      class="c0"
      display="flex"
      role="article"
      width="100%"
    >
      <div
        aria-label="store name"
        class="c1"
        display="flex"
        height="15rem"
        role="figure"
      >
        <div
          class="c2"
        >
          <img
            alt="store name"
            class="lazyload"
            data-sizes="auto"
            data-srcset="S3_BUCKET/img-300w.png 300w, S3_BUCKET/img-600w.png 600w"
            src="S3_BUCKET/img-600w.png"
            srcset="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
          />
        </div>
      </div>
      <div
        aria-label="store name logo"
        class="c3"
        height="15rem"
        role="figure"
        width="15rem"
      >
        <div
          style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
        >
          <div
            style="box-sizing: border-box; display: block; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTUwIiBoZWlnaHQ9IjE1MCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
              style="max-width: 100%; display: block;"
            />
          </div>
          <img
            alt="store name logo"
            decoding="async"
            src="/_next/image?url=%2Fimg-300w.png&w=640&q=75"
            srcset="/_next/image?url=%2Fimg-300w.png&w=256&q=75 1x, /_next/image?url=%2Fimg-300w.png&w=384&q=75 2x, /_next/image?url=%2Fimg-300w.png&w=640&q=75 3x"
            style="visibility: visible; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </div>
      </div>
      <div
        class="c4"
        display="flex"
        width="100%"
      >
        <p
          class="c5"
        >
          store name
        </p>
        <div
          class="c6"
          display="flex"
          width="100%"
        >
          <div
            aria-label="distance in minutes"
            class="c7"
            display="flex"
          >
            <svg
              aria-label="location"
              class="c8"
              fill="currentColor"
              height="1em"
              role="img"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M444.52 3.52L28.74 195.42c-47.97 22.39-31.98 92.75 19.19 92.75h175.91v175.91c0 51.17 70.36 67.17 92.75 19.19l191.9-415.78c15.99-38.39-25.59-79.97-63.97-63.97z"
              />
            </svg>
            <p
              aria-label="time"
              class="c9"
              color="grayQuaternary"
            >
              10
              min
            </p>
          </div>
          <div
            class="c10"
            display="inline-flex"
          >
            <div
              class="c11"
              display="inline-flex"
            >
              <p
                class="c12"
                color="textInverted"
              >
                open
              </p>
            </div>
            <p
              class="c9"
              color="grayQuaternary"
            >
              until 10h
            </p>
          </div>
          <div
            aria-label="like counter"
            class="c7"
            display="flex"
          >
            <svg
              aria-label="likes"
              class="c13"
              fill="currentColor"
              height="1em"
              role="img"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
              />
            </svg>
            <p
              class="c9"
              color="grayQuaternary"
            >
              10
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
